### ISSUE TITLE
Two IRIs are defined in different namespaces in a way that, taken together, violate OWL 2 DL

### ISSUE THREAD

**Issue by ajnelson-nist** [qudt/qudt-public-repo#826]
Hello,

I found that on loading a plurality of QUDT Turtle files from today's `main` branch (Git state `f2711c5`) that some concept IRIs are defined in multiple files, at points in manners inconsistent with OWL 2 DL - specifically, with OWL 2 DL's requirements on properties being exactly one of an annotation property, datatype property, or object property ([Syntax Section 5.8.1](https://www.w3.org/TR/2012/REC-owl2-syntax-20121211/#Typing_Constraints_of_OWL_2_DL)).

This appears to be related to Issue #289 and one or two hops up its chain of related Issues.

The two instances I saw are:

* `qudt:rdfsDatatype` is defined in these two places, one as a `owl:DatatypeProperty`, one as a `owl:ObjectProperty`:
   * [The datatype file](https://github.com/qudt/qudt-public-repo/blob/f2711c5/schema/SCHEMA_QUDT-DATATYPE-v2.1.ttl#L6144-L6148)
   * [The core file](https://github.com/qudt/qudt-public-repo/blob/f2711c5/schema/SCHEMA_QUDT-v2.1.ttl#L2965-L2966)
* `dcterms:description` is defined in these two places, one as a `owl:DatatypeProperty`, one as both an `owl:DatatypeProperty` and `owl:AnnotationProperty`:
   * [The datatype file](https://github.com/qudt/qudt-public-repo/blob/f2711c5/schema/SCHEMA_QUDT-DATATYPE-v2.1.ttl#L35-L37)
   * [The core file](https://github.com/qudt/qudt-public-repo/blob/f2711c5/schema/SCHEMA_QUDT-v2.1.ttl#L41-L42)

My guess at a resolution for the `dcterms:description` issue is that it should be only an OWL Annotation Property, because Dublin Core only provides RDFS definitions, not OWL definitions.  (I'd be happy to be corrected on that if one has come out since I last looked!)

## Discovery mechanism

_Disclaimer: Participation by NIST in the creation of the documentation of mentioned software is not intended to imply a recommendation or endorsement by the National Institute of Standards and Technology, nor is it intended to imply that any specific software is necessarily the best available for the purpose._

I found these by running [this shapes graph](https://github.com/ucoProject/UCO/blob/1.2.0/ontology/owl/owl.ttl) against one graph that had these files loaded:

* `schema/SCHEMA-FACADE_QUDT-v2.1.ttl`
* `schema/SCHEMA_QUDT-v2.1.ttl`
* `schema/SCHEMA_QUDT-DATATYPE-v2.1.ttl`
* `schema/SHACL-SCHEMA-SUPPLEMENT_QUDT-v2.1.ttl`

(That graph also raises some notes about `rdf:List` usage, but I deactivated that shape for the purposes of isolating these multi-property-type issues.)

**Comment by steveraysteveray** [comment]
Thanks for your thorough investigation of this! It seems to me that qudt:rdfs:Datatype should be declared as an owl:DatatypeProperty.

Regarding dcterms:description, I'm thinking it should be an owl:DatatypeProperty as well, because as I recall, owl:AnnotationProperties cannot be used in OWL restriction classes. Is that your understanding?

**Comment by ajnelson-nist** [comment]
> Thanks for your thorough investigation of this! It seems to me that qudt:rdfsDatatype should be declared as an owl:DatatypeProperty.

I'll defer to you on this, I'm not familiar with the intended usage yet.
 
> Regarding dcterms:description, I'm thinking it should be an owl:DatatypeProperty as well, because as I recall, owl:AnnotationProperties cannot be used in OWL restriction classes. Is that your understanding?

I agree on the latter conclusion.  Scanning over the [OWL 2 RDF syntax document](https://www.w3.org/TR/2012/REC-owl2-mapping-to-rdf-20121211/), which has a strict pattern-matching requirement spelled in the line just before section 4, `owl:AnnotationProperty` never appears in an `owl:Restriction`.  (See esp. Tables 13 and 1, scanning for "owl:Restriction".)

My current belief is that `dcterms:description` needs to be an OWL Annotation Property.  Neither the RDFS definition for `dcterms:description` nor its superproperty `dc:description` include `rdfs:range` prescription, so you could link an object with `dcterms:description` and still be conformant with Dublin Core Terms.
Some other properties, like `dcterms:dateSubmitted`, have a prescribed range `rdfs:Literal`, so that property _could_ be an OWL Datatype Property; but, there would need to be a consensus built on that, and friction could arise if two RDF ecosystems collide that disagree on annotation vs. datatype.
With all the above, I think `dcterms:description a owl:AnnotationProperty .` would be correct.  If there's a desire to constrain its range to be literal-valued, a SHACL property shape could be used to constrain the range node kind to `sh:Literal` within the context of usage within QUDT and QUDT adopters.

**Comment by steveraysteveray** [comment]
I would agree with you on the dcterms:description, but for QUDT we do impose some cardinality constraints on the relation. That is not an issue for the SHACL schema for QUDT, but it is for the OWL schema (we try to maintain both schemas). Specifically an owl:maxCardinality of 1 for our root class, qudt:Concept.

I'm tempted to try it out with dcterms:description as an owl:DatatypeProperty and see if it breaks the loading into Protege. Please let me know if you think this is a bad idea.

Also, I'm impressed with the shapes graph you pointed to, and plan to look into it more deeply. Thanks for the pointer.

**Comment by ajnelson-nist** [comment]
> I'm tempted to try it out with dcterms:description as an `owl:DatatypeProperty` and see if it breaks the loading into Protege. Please let me know if you think this is a bad idea.

I think, within the context of only QUDT within this repository, you will not encounter any problems picking one or the other, aside from wanting the cardinality restriction specified in OWL.

I also think, in a broader context looking to a plurality of QUDT adopters that I will just assume for the purposes of this comment, that you will open yourself up to trouble from an OWL transitive import closure stumbling on the same property-class conflict that I did, if you tried to assert `owl:DatatypeProperty`.  Let's say some QUDT adopter went with the "maximally-allowed, least-restricted" mindset I hinted at previously.  ("Nothing says I can't link an object...welp, I guess it's gotta be anno.")  That adopter would always hit the DP-AP conflict firing up a tool that needs OWL 2 DL conformance.

I'm not asserting these conflicts exist today, but I do believe they're possible the more "inheriting" a downstream data model becomes.

**Comment by steveraysteveray** [comment]
I'm going to have to think more deeply about this. (I'm also hoping most new adopters will be using SHACL rather than OWL, where this problem goes away).

Meanwhile, I noticed another thing in your original post:

schema/SCHEMA-FACADE_QUDT-v2.1.ttl
schema/SCHEMA_QUDT-v2.1.ttl
schema/SCHEMA_QUDT-DATATYPE-v2.1.ttl
schema/SHACL-SCHEMA-SUPPLEMENT_QUDT-v2.1.ttl

This combination of files should not happen. Either a user would load the web-resident version of SCHEMA-FACADE_QUDT-v2.1.ttl (i.e. [this](http://qudt.org/2.1/schema/facade/qudt)), that imports the other files but not the SHACL supplement file, or

...a user loads the repo version of SCHEMA-FACADE_QUDT-v2.1.ttl that imports schema/shacl/SCHEMA_QUDT_NoOWL-v2.1.ttl rather than schema/SCHEMA_QUDT-v2.1.ttl. 

This arrangement is our attempt to keep the LinkedData/OWL people happy with web-loading, and the SHACL people happy with repo-loading, described [here](https://github.com/qudt/qudt-public-repo#protege-users).

How did you come to group the files the way you did?

**Comment by ajnelson-nist** [comment]
> How did you come to group the files the way you did?

Naïvely.  I took everything that looked like the current files and piled them into one.  Apologies that I missed documentation prescribing which files to import.

I remember intending to note this was a naïve file grouping.  But as I was writing things up, the fundamental issue turned out to be independent of whether I grouped files incorrectly, so I forgot that note.

Thank you for the prescriptions, though!

**Comment by steveraysteveray** [comment]
I agree - the problem exists regardless. I thought I'd try to learn from your experience to better guide our users.

Meanwhile, I tried running your owl.ttl file on the repository and I see the List-shape violation. I'll have to go back and read up on why OWL insists the subject of an rdf:first needs to be a blank node. Unexpected.

**Comment by ajnelson-nist** [comment]
As an aside: I suggest the List shape discussion be left as out of scope of resolving this Issue.

I noted in my first post that I deactivated that shape as part of the review.  It was a shape that, in discussions with the committee that maintains that file, I'd expressed I was unhappy to have found necessary to write, but my review of the OWL spec was telling me it seemed necessary.

I'm close to proposing revising that shape.  I've realized, in the while that shape has been operational, that there are differing concerns in OWL review of `rdf:List`.  The [OWL-RDF syntax document](https://www.w3.org/TR/2012/REC-owl2-mapping-to-rdf-20121211/) makes specific requirements on `T(SEQ y1 ... yn)` inducing blank nodes (Table 1).  What I missed was a scope: `T(SEQ y1 ... yn)` seems to only apply as a first step from certain OWL predicates.  I missed that this would not apply for other constructs, e.g. SKOS's [`skos:memberList`](https://www.w3.org/TR/skos-reference/#L3312) (and I recall SKOS reports OWL 2 DL conformance), or DASH's [`dash:dateOrDateTime`](https://datashapes.org/dash.html#DateOrDateTime).

I need to find a few other citations around `rdf:List` handling, specifically a line I recall reading where somewhere between RDF, RDF Schema, and OWL, there is a requirement that `rdf:List`s be strictly linear.  But by the end, that specific list shape will raise fewer reported issues.  The SHACL usage in QUDT of `rdf:List` is, I suspect, being flagged as a false-positive issue by that UCO OWL shape.  It's down to a matter of hopping between the four specifications to confirm everything.

Recapping, specifically to QUDT - I have no remarks on QUDT's current usage of `rdf:List`.  I'll file notes if I think there are any.  But needing to repeat lists because of a blank node restriction is a known maintenance concern.
